src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -340,9 +458,11 @@ struct UpdateComponentsAndEdges
   {
     auto const n = _out_edges.extent(0);
 
-    int next_component = _labels(_out_edges(component - n + 1).target);
+    assert(_out_edges(component - n + 1).target() != INT_MAX);
+    int next_component = _labels(_out_edges(component - n + 1).target());
+    assert(_out_edges(next_component - n + 1).target() != INT_MAX);
```

Comment:
I think the debug assertions can be removed. I don't see how `INT_MAX` is special.